src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs

```diff
@@ -510,6 +559,20 @@ public void AddClaims(IEnumerable<Claim> claims)
             }
         }
 
+        /// <summary>
+        /// Adds claims from dictionary.
+        /// </summary>
+        /// <param name="claimsCollection"> A dictionary of claims.</param>
+        /// <remark> If a key is already present in target dictionary, its value is overridden by the value of the key in claimsCollection.</remark>
+        internal void MergeDictionaryClaims(IDictionary<string, object> claimsCollection)
```

Comment:
This name is misleading. It doesn't look merging is done

Reply:
Right. Its more like adding.